src/frequenz/sdk/microgrid/component/battery_utils.py

```diff
@@ -0,0 +1,60 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Util functions for battery."""
+
+import logging
+from typing import Dict, Optional, Set
+
+from .. import get as get_microgrid
+from . import ComponentCategory
+
+_logger = logging.getLogger(__name__)
+
+
+def find_adjacent_inverter_id(battery_id: int) -> Optional[int]:
+    """Find inverter adjacent to this battery.
+
+    Args:
+        battery_id: battery id adjacent to the wanted inverter
+
+    Note:
+        Method doesn't check if given id belongs to the battery.
+
+    Returns:
+        Id of the adjacent inverter. Return None if battery has not adjacent inverter.
+    """
+    graph = get_microgrid().component_graph
+    try:
+        inverter_id = next(
+            (
+                comp.component_id
+                for comp in graph.predecessors(battery_id)
+                if comp.category == ComponentCategory.INVERTER
+            ),
+            None,
+        )
+    except KeyError as err:
+        _logger.warning(err)
```

Comment:
> Do we really want to log an error each time someone tries to find an inverter for a battery and it's not there? This error is also duplicated when the battery_inverter_mapping() function is used.

This error tells there is no component with this id in component graph. So yes it would be nice to print it :D

> Also, if we keep it, I would write a more descriptive error instead of printing the err object raw.

It is already descriptive enough in the ComponentGraph.

> BTW, is a microgrid in a valid state if it has batteries without inverters? Because if it's not maybe this should be an error instead.

It is valid state to have inverter without battery. Few months ago we added it.
I think component graph allows for battery without inverter, so it is valid state, too. (I can check it if necessary) 
Anyway these requirements can change in the future, so I prefer to check it here. It is one-time check and cost me nothing.

Reply:
True, thanks for noting that!
I will change it to info log :)